lib/_http_client.js

```diff
@@ -393,6 +391,12 @@ function socketCloseListener() {
   }
 }
 
+function emitError(err) {
```

Comment:
Only a nit that can be ignore. I would pass `req` as an argument so we can use `process.nextTick(emitError, this, err)` above.